github/search.go

```diff
@@ -182,15 +183,14 @@ func (s *SearchService) search(searchType string, query string, opt *SearchOptio
 		return nil, err
 	}
 
-	if searchType == "commits" {
-		// Accept header for search commits preview endpoint
-		req.Header.Set("Accept", mediaTypeCommitSearchPreview)
-	}
-
-	if opt != nil && opt.TextMatch {
-		// Accept header defaults to "application/vnd.github.v3+json"
-		// We change it here to fetch back text-match metadata
-		req.Header.Set("Accept", "application/vnd.github.v3.text-match+json")
+	switch {
+		case opt != nil && opt.TextMatch:
+			// Accept header defaults to "application/vnd.github.v3+json"
+			// We change it here to fetch back text-match metadata
+			req.Header.Set("Accept", "application/vnd.github.v3.text-match+json")
+		case searchType == "commits":
+			// Accept header for search commits preview endpoint
+			req.Header.Set("Accept", mediaTypeCommitSearchPreview)
```

Comment:
Two things here.

1. Add a `// TODO: remove custom Accept header when this API fully launches.` comment.

2. Now that these cases are in the same switch, it's a little easier to reason about them. I think we should give the `searchType == "commits"` higher priority than `opt != nil && opt.TextMatch`. It may not even be valid to set `opt.TextMatch: true` for a commit search, but if someone does, it would fail to return any results at all because the custom accept header wouldn't be set.

    So, let's be safe and put the `searchType == "commits"` case on top so that the API preview header is always set when doing a commit search.